command/ca/provisioner/provisioner.go

```diff
@@ -101,38 +108,68 @@ type crudClient interface {
 }
 
 func newCRUDClient(cliCtx *cli.Context, cfgFile string) (crudClient, error) {
-	// os.Stat("") probably returns os.ErrNotExist, but this behavior is
-	// undocumented so we'll handle this case separately.
-	if cfgFile == "" {
-		return cautils.NewAdminClient(cliCtx)
+	caURL, err := flags.ParseCaURLIfExists(cliCtx)
+	if err != nil {
+		return nil, err
+	}
+	if caURL == "" {
+		return nil, errs.RequiredFlag(cliCtx, "ca-url")
+	}
+	root := cliCtx.String("root")
+	if root == "" {
+		root = pki.GetRootCAPath()
+		if _, err := os.Stat(root); err != nil {
+			return nil, errs.RequiredFlag(cliCtx, "root")
+		}
+	}
+	rootCAs, err := x509util.ReadCertPool(root)
+	if err != nil {
+		return nil, fmt.Errorf("error generating cert pool: %w", err)
 	}
 
-	_, err := os.Stat(cfgFile)
-	switch {
-	case errors.Is(err, os.ErrNotExist):
-		return cautils.NewAdminClient(cliCtx)
-	case err == nil:
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: false,
+		MinVersion:         tls.VersionTLS12,
+		RootCAs:            rootCAs,
+	}
+	tr := &http.Transport{
+		IdleConnTimeout: 10 * time.Second,
+		TLSClientConfig: tlsConfig,
+	}
+
+	client := &http.Client{Transport: tr}
+	// TODO: Replace this logic with url.JoinPath once go1.18 is no longer supported.
+	u, err := url.Parse(caURL)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing caURL: %w", err)
+	}
+	u.Path = path.Join(u.Path, "admin/admins")
```

Comment:
It's probably better to use the following:
```suggestion
	u = u.ResolveReference(&url.URL{Path: "/admin/admins"})
```